mobile-ui/src/main/java/com/playone/mobile/ui/firebase/v1/PlayoneFirebaseV1.kt

```diff
@@ -1,70 +1,192 @@
 package com.playone.mobile.ui.firebase.v1
 
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
 import com.google.firebase.database.DatabaseReference
-import com.playone.mobile.data.model.NotificationPayloadEntity
-import com.playone.mobile.data.model.PlayoneEntity
-import com.playone.mobile.data.model.UserEntity
-import com.playone.mobile.data.repository.PlayoneRemote
+import com.google.firebase.database.Query
 import com.playone.mobile.ext.isNotNull
 import com.playone.mobile.remote.PlayoneFirebase
+import com.playone.mobile.remote.model.PlayoneModel
+import com.playone.mobile.remote.model.UserModel
 import com.playone.mobile.ui.firebase.ext.addListenerForSingleValueEvent
 
 class PlayoneFirebaseV1(
     private val dbReference: DatabaseReference
-) : PlayoneFirebase(), PlayoneRemote {
-    fun playoneDataSnapshot() {
-        dbReference.child(GROUPS).addListenerForSingleValueEvent {
-            onDataChange = {
-                it.takeIf { it.isNotNull() }?.let {}
-            }
-            onCancelled = { }
-        }
+) : PlayoneFirebase() {
+
+    /**
+     * Retrieve the [List] of the [PlayoneModel] from the firebase server. We'll use
+     * the callback function for returning the value back to.
+     *
+     * @param userId user id.
+     * @param callback a function for fetching a [List] of the [PlayoneModel].
+     * @param errorCallback a function for getting a error when retrieving the data.
+     */
+    override fun getPlayoneList(
+        userId: Int,
+        callback: PlayoneCallback<List<PlayoneModel>>,
+        errorCallback: FirebaseErrorCallback
+    ) = if (0 < userId) {
+        playoneDataSnapshot(callback, errorCallback, ::snap2PlayoneList)
+    }
+    else {
+        userDataSnapshot(userId.toString(),
+                         {},  // This is redundant anonymous function for running strategy function.
```

Comment:
Please follow this https://android.github.io/kotlin-guides/style.html#continuation-indent